packages/css-data/src/property-parsers/transition.ts

```diff
@@ -0,0 +1,105 @@
+import * as csstree from "css-tree";
+import type {
+  InvalidValue,
+  LayersValue,
+  TupleValue,
+  TupleValueItem,
+  Unit,
+} from "@webstudio-is/css-engine";
+
+const cssTreeTryParseValue = (input: string) => {
+  try {
+    const ast = csstree.parse(input, { context: "value" });
+    return ast;
+  } catch {
+    return undefined;
+  }
+};
+
+export const parseTransition = (
+  transition: string
+): LayersValue | InvalidValue => {
+  let tokenStream = transition.trim();
+  tokenStream = tokenStream.endsWith(";")
+    ? tokenStream.slice(0, -1)
+    : tokenStream;
+
+  const cleanupKeywords = ["transition:"];
+
+  for (const cleanupKeyword of cleanupKeywords) {
+    tokenStream = tokenStream.startsWith(cleanupKeyword)
+      ? tokenStream.slice(cleanupKeyword.length).trim()
+      : tokenStream;
+  }
+
+  const cssAst = cssTreeTryParseValue(tokenStream);
+
+  if (cssAst === undefined) {
+    return {
+      type: "invalid",
+      value: transition,
+    };
+  }
+
+  const parsed = csstree.lexer.matchProperty("transition", cssAst);
+  if (parsed.error) {
+    return {
+      type: "invalid",
+      value: transition,
+    };
+  }
+
+  const layers: TupleValue[] = [];
+
+  csstree.walk(cssAst, (node) => {
```

Comment:
Imo underlying algorithm should be simple - simple.
It's just splitting children array on Operator `,` but too much logic for this simple thing.
like this comparisons `children.last === child` looks useless

Reply:
The first iteration of design has individual display like, so the parser was little complicated to pick everything. Looks like the design is changed now, and the parser also can be just simple one now.